pkg/controller/restore_controller.go

```diff
@@ -324,8 +324,8 @@ func (c *restoreController) validateAndComplete(restore *api.Restore, pluginMana
 		selector := labels.SelectorFromSet(labels.Set(map[string]string{
 			velerov1api.ScheduleNameLabel: restore.Spec.ScheduleName,
 		}))
-
-		backups, err := c.backupLister.Backups(c.namespace).List(selector)
+		// NOTE: This assumes that Backups and Restores always reside in the same namespace.
+		backups, err := c.backupLister.Backups(restore.Namespace).List(selector)
```

Comment:
so for clarification: When we do a restore, we have to:
* restore it to the same cluster id (and because of that same namespace)
* restore it manually (move the backup crd to another namespace or sth like that!?)

However, old behaviour was the same for the velero namespace right.

Reply:
Yes, actually as long as the Backup, the BackupStorageLocation and the Restore are in the same namespace (not necessarily the cluster namespace, could potentially be anyone as long as the BackupStorageLocation points to a valid backup in S3) everything's good. But it makes the most sense to use the same cluster namespace as before when restoring.